// Copyright (C) 2016 Nippon Telegraph and Telephone Corporation.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//    http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
// implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

package knet;

message NetIF {
    bytes name = 1;
    int64 port = 2;
}

message InitKNETRequest {
    int64 unit = 1;
}

message InitKNETResponse {
}

message AddKNETRequest {
    int64 unit = 1;
    NetIF netif = 2;
}

message AddKNETResponse {
    int64 id = 1;
    int64 filter_id = 2;
}

message DeleteKNETRequest {
    int64 unit = 1;
    int64 id = 2;
    int64 filter_id = 3;
}

message DeleteKNETResponse {
}

message ListKNETRequest {
    int64 unit = 1;
}

message ListKNETResponse {
    repeated NetIF list = 1;
}
